javalib/src/main/scala/java/util/regex/Pattern.scala

```diff
@@ -107,10 +106,12 @@ object Pattern {
     compile(regex).matcher(input).matches
 
   def quote(s: String): String = {
-    val original = StringPart(s)
-    val quoted   = StringPart.stackalloc
+    val original, quoted = stackalloc[cre2.string_t]
+    toRE2String(s, original)
     cre2.quoteMeta(quoted, original)
-    quoted.toString
+    val res = fromRE2String(quoted)
+    free(original.data.cast[Ptr[Byte]])
```

Comment:
`fromRE2String` calls `toCString`, which uses `malloc` to allocate the memory. It means we must free it afterwards.